doc/concepts/pipeline-concepts/datum/glob-pattern.md

```diff
@@ -0,0 +1,128 @@
+repository for data processing in the most optimal way.
+
+Each time Pachyderm detects new commits in your repository, it starts the
+pipeline and processes the whole datum. Therefore, there might be
+a significant difference between processing the whole repository
+or one directory in the repository. Consider this
+behavior when you design your pipeline.
+
+Defining how your data is spread among workers is one of
+the most important aspects of distributed computation and is
+the fundamental idea around concepts, such as Map and Reduce.
+
+Instead of confining users to data-distribution patterns,
+such as Map, that splits everything as much as possible, and
+Reduce, that groups all the data, Pachyderm
+uses glob patterns to provide incredible flexibility to
+define your data distribution.
+
+You can configure a glob pattern for each PFS input in
+the input field of a pipeline specification. Pachyderm detects
+this parameter and divides the input data into
+individual *datums* to process them independently according
+to the provided glob pattern.
+
+Think of the PFS input repository as a filesystem where
+the glob pattern is applied to the root of the
+filesystem. The files and directories that match the
+glob pattern are considered datums.
+
+Pachyderm accepts the `/` and `*` indicators as
+glob patterns.
+
+The following are examples of glob patterns that you can define:
+
+* `/` — Pachyderm processes the whole repository as a
+  single datum and sends the data to a single worker node.
+* `/*` — Pachyderm processes each top-level filesystem
+  object, such as a file or directory, in
+  the root directory as a separate datum. For example,
+  if you have a repository with ten files in it and no
+  directory structure, Pachyderm handles each file as a
+  single datum, similar to a map operation.
+* `/*/*` — Pachyderm processes each filesystem object
+  in each subdirectory as a separate datum.
+
+<!-- Add the ohmyglob examples here-->
+
+The following text is an extract from a pipeline specification.
+It shows how you can define glob pattern for individual PFS
+inputs:
+
+```
+"input": {
+    "cross" or "union": [
+        {
+            "pfs": {
+                "repo": "string",
+                "glob": "string",
+            }
+        },
+        {
+            "pfs": {
+                "repo": "string",
+                "glob": "string",
+            }
+        },
+        etc...
+    ]
+ }
+ ```
+
+For example, if you have two workers and define two datums,
+Pachyderm sends one datum to each worker. If you have more
+datums than workers, Pachyderm queues datums and sends
+them to workers as they finish processing previous datums.
+
+## Example of Defining Datums
+
+For example, you have the following directory:
+
+```bash
+/California
+   /San-Francisco.json
+   /Los-Angeles.json
+   ...
+/Colorado
+   /Denver.json
+   /Boulder.json
+   ...
+...
+```
+
+Each top-level directory represents a United State (US) with
```

Comment:
> Each top-level directory represents a United State (US) with
`json` files in it that represent cities of that state.

Lol, no one has ever referred to a single state as a "United State (US)". hehe. First, I think this section just goes away because I've moved it up further. Second, if you are going to keep it, you would need to instead say,

"Each top-level directory is a US state with data
(a `json` file) for each city in that state."